docs/root/intro/arch_overview/upstream/circuit_breaking.rst

```diff
@@ -13,7 +13,12 @@ configure and code each application independently. Envoy supports various types
 
 * **Cluster maximum connections**: The maximum number of connections that Envoy will establish to
   all hosts in an upstream cluster. If this circuit breaker overflows the :ref:`upstream_cx_overflow
-  <config_cluster_manager_cluster_stats>` counter for the cluster will increment.
+  <config_cluster_manager_cluster_stats>` counter for the cluster will increment. All connections,
+  whether active or draining, count against this limit. Even if this circuit breaker has overflowed,
+  Envoy will ensure that a host selected by cluster load balancing has at least one connection
+  allocated. This has the implication that the :ref:`upstream_cx_active
+  <config_cluster_manager_cluster_stats>` count for a cluster may be as high as
+  `cluster maximum connections + number of endpoints in a cluster`.
```

Comment:
Thinking about this I think this is actually `cluster maximum connections + (number of connection pools) * (number of endpoints in a cluster)`, where each connection pool is identified by its HTTP protocol, routing priority and socket/TLS options. This number would be limited by the max connection pool circuit breaker limit. 

This is where the logic for determining whether a new conn pool is required: https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cluster_manager_impl.cc#L1279-L1329

In reality I doubt most people are using more than 2 (one for HTTP/2 and one for HTTP/1.1), with the vast majority only using one. 

Not sure how in depth we want to get here, I've found that explaining the concept of multiple connection pools can be tricky, since the thing that leads to a large number of conn pools (dynamic socket options) is a fairly esoteric use case.

Reply:
May be not needed to go much in depth to explain implementation details, but worth mentioning that users could see  upstream_cx_active > cluster maximum connections. Out team got super confused with circuit breaker stats recently.